### PR TITLE
Fix/session_signout

### PR DESCRIPTION
**Pull request by mitcheman**
Main changes are moving the RefreshToken function out of a utility function the server side is calling when requesting models/sending chat messages and to the /api/auth/[nextauth] route. 

The new logic uses the JWT callback in the nextauth route on Initial Signin. That gets the initial accessToken/refreshToken/AccessTokenExpireation from Azure AD and that is added to the token that gets passed to the Session callback on the client side. The session information is then appended with that information via the session callback on the client. 

Now on an accessToken expiration of 1 hour (default for azure AD and not sure it can be changed) - all authentication requests are passed to the JWT callback via the middleware. If the token that is passed into the JWT callback has expired it requests a new one via the RefreshAccessToken function, passing in its refreshToken (with an expiration of 30 days). 

Basically now the refreshToken logic is in the same function that handles the initial signIn instead of on a protected server route that needs authentication. 

There is also now a UseEffect hook on the homePage that checks for refreshToken errors and pushes the user to sign in again if it could not refresh the token.

### CODE REVIEW

**Review by blaze-burgess-msf — APPROVED** [review]
Judging mostly based on observed functionality, but the code looked good as well. Approved!
